ruby/lesson_9/code_smell_2/baitap_codesmell.rb

```diff
@@ -0,0 +1,92 @@
+class OrderProcessor
+
+  def process_order(order)
+    customer = order.customer
+    items = order.items
+
+
+    order.existence_check()
+    total_price = order.total_order_value()
+    total_price = order.discount_apply(total_price)
+    # Tạo hóa đơn
+    invoice = create_invoice(customer, items, total_price)
+    # Gửi email xác nhận
+    send_confirmation_email(customer, invoice)
+
+  end
+
+  private
+
+  def create_invoice(customer, items, total_price)
+    # Tạo hóa đơn và lưu vào cơ sở dữ liệu
+    # ...
+  end
+
+  def send_confirmation_email(customer, invoice)
+    # Gửi email xác nhận cho khách hàng
+    # ...
+  end
+end
+
+
+class Order
+  attr_accessor :customer, :items
+  def initialize(customer, items)
+    @customer = customer
+    @items = items
+  end
+
+  # Kiểm tra có mặt hàng hay không
+  def existence_check()
+    raise "Không có mặt hàng nào!" if @items.empty?
+    # Kiểm tra tồn kho
+    @items.each do |item|
+      item.item_inventory_check()
+    end
+  end
+
+  # Tính tổng giá trị đơn hàng
+  def total_order_value()
+    total_price = 0
+    @items.each do |item|
+      total_price += item.price * item.quantity
```

Comment:
ở đây đang sử dụng price và quantity của item để tính toán (feature envy) --> đưa vào 1 method tính price ở Item class